camayoc/api.py

```diff
@@ -121,30 +126,70 @@ def __init__(self, response_handler=None, url=None):
         else:
             self.response_handler = response_handler
 
+        if self.authenticate:
+            self.login()
+
+    def login(self):
+        """Login to the server to receive an authorization token."""
+        self.authenticate = True
```

Comment:
I would not recommend reusing the flag to indicate if should be authenticated with the state of being authenticated or not. It seems that having another flag like `_authenticated` will fit better and allow control the internal state of being or not authenticated.

With that the client will be able to tell if it is supposed or not to be authenticated and if it is authenticated or not.